### PR TITLE
SIKE/CMake: recognize both x64+BMI2 and x64+BMI2+ADX

### PR DESCRIPTION
**Pull request by jschanck**
While looking into https://github.com/open-quantum-safe/profiling/issues/47 I noticed that the bugfix in #963 was slightly too restrictive.

SIKE provides assembly routines for both x64+BMI2 and x64+BMI2+ADX systems, but it requires different #defines in the two cases. Prior to #963 we were setting the wrong #defines for x64+BMI2, which caused #961. I assumed that SIKE only supported x64+BMI2+ADX in #963, and I disabling support for x64+BMI2.

This PR re-enables support for x64+BMI2 and sets the right #defines for both configurations.

### CODE REVIEW

File: src/kem/sike/CMakeLists.txt [hunk 1]
```diff
@@ -82,8 +83,11 @@ if(OQS_ENABLE_KEM_sike_p434 OR
         target_compile_definitions(sike PRIVATE _GENERIC_ _X86_)
     elseif(ARCH_X86_64)
         target_compile_definitions(sike PRIVATE _AMD64_)
-        if(OQS_USE_ADX_INSTRUCTIONS AND NOT OQS_DIST_BUILD)
-            target_compile_definitions(sike PRIVATE _MULX_ _ADX_)
+        if(OQS_USE_BMI2_INSTRUCTIONS)
```
Comment (baentsch) [inline src/kem/sike/CMakeLists.txt:86]: Woudn't it be more logical/readable to have the two options be independent? Or would the code break if only ADX is present but not BMI2?

Comment (jschanck) [reply src/kem/sike/CMakeLists.txt:86]: The two options are not independent. Every machine with ADX instructions has BMI2 instructions.

File: src/kem/sike/CMakeLists.txt [hunk 2]
```diff
@@ -92,7 +96,8 @@ if(OQS_ENABLE_KEM_sike_p434 OR
             if(OQS_ENABLE_KEM_sike_p751 OR OQS_ENABLE_KEM_sidh_p751 OR OQS_ENABLE_KEM_sike_p751_compressed OR OQS_ENABLE_KEM_sidh_p751_compressed)
                 target_compile_definitions(sike PRIVATE USE_SIKEP751_ASM)
             endif()
-            if(OQS_USE_ADX_INSTRUCTIONS AND NOT OQS_DIST_BUILD)
+            if(OQS_USE_BMI2_INSTRUCTIONS)
```
Comment (baentsch) [inline src/kem/sike/CMakeLists.txt:99]: The switch from checking for ADX to BMI2 I understand (if the latter also includes/implies the former or the code builds regardless of ADX presence) but leaving away `OQS_DIST_BUILD` here seems to imply that the resultant code will work regardless of CPU features found, no? Asked another way, what code difference results for SIKE with `OQS_DIST_BUILD` ON or OFF?

Comment (jschanck) [reply src/kem/sike/CMakeLists.txt:99]: The OQS_USE_XXX_INSTRUCTIONS are only set when OQS_DIST_BUILD=OFF.

https://github.com/open-quantum-safe/liboqs/blob/efd7b1926ac485bd2c284986cb94e51f4d50450c/.CMake/compiler_opts.cmake#L51-L53

Comment (baentsch) [reply src/kem/sike/CMakeLists.txt:99]: OK, I was confused how then run-time feature detection works when OQS_DIST_BUILD=ON given this (example Kyber, other places equivalent) code: https://github.com/open-quantum-safe/liboqs/blob/efd7b1926ac485bd2c284986cb94e51f4d50450c/src/kem/kyber/kem_kyber_512.c#L44-L56 . But this https://github.com/open-quantum-safe/liboqs/blob/efd7b1926ac485bd2c284986cb94e51f4d50450c/.CMake/alg_support.cmake#L178-L180 in combination with this https://github.com/open-quantum-safe/liboqs/blob/efd7b1926ac485bd2c284986cb94e51f4d50450c/CMakeLists.txt#L33-L35 makes it clear again. So case closed. Thanks.

**Review by baentsch — APPROVED** [review]
After the explanations, LGTM.

**Pull request merged**